.github/workflows/master.yml

```diff
@@ -252,13 +267,86 @@ jobs:
           export MLFLOW_HOME=$(pwd)
           pytest tests/models
 
+  pipelines:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      # - name: Set up mlflow-pipelines-example-private submodule deploy key
+      #   env:
+      #     SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+      #   run: |
+      #       ssh-agent -a $SSH_AUTH_SOCK > /dev/null
+      #       ssh-add - <<< "${{ secrets.REPO_PRIVATE_SSH_KEY }}"
+      # - name: Initialize submodules
+      #   env:
+      #     SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+      #   run: |
+      #     git submodule update --init
+      - uses: ./.github/actions/setup-python
+        with:
+          python-version: "3.7"
+      - name: Install dependencies
+        env:
+          INSTALL_SMALL_PYTHON_DEPS: true
+          INSTALL_LARGE_PYTHON_DEPS: false
+        run: |
+          source ./dev/install-common-deps.sh
+          pip install -e .[pipelines]
+          pip install pyspark
+      - name: Run tests
+        run: |
+          export MLFLOW_HOME=$(pwd)
+          pytest tests/pipelines
+
+  pipelines-windows:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - uses: actions/setup-python@v3
+        with:
+          python-version: "3.7"
+      # - name: Initialize submodules
+      #   run: |
+      #     # Start Windows SSH agent and add private deploy key for
+      #     # mlflow-pipelines-example-private submodule
+      #     eval `ssh-agent.exe`
+      #     ssh-add - <<< "${{ secrets.REPO_PRIVATE_SSH_KEY }}"
+      #     # Add github.com to known hosts
+      #     mkdir -p ~/.ssh
+      #     ssh-keyscan -t ed25519 github.com >> ~/.ssh/known_hosts
+      #     # Initialize submodules
+      #     git submodule update --init
+      - name: Install python dependencies
+        run: |
+          pip install -r requirements/test-requirements.txt
+          pip install --no-dependencies tests/resources/mlflow-test-plugin
+          pip install -e .[extras]
+          pip install -e .[pipelines]
+          pip install pyspark
```

Comment:
```suggestion
          # TODO: Unpin once Delta supports Spark 3.3 (https://github.com/delta-io/delta/issues/1217)
          pip install 'pyspark<3.3'
```

Reply:
Pinning pypsark should fix this test failure: https://github.com/mlflow/mlflow/runs/7017120343?check_suite_focus=true#step:5:240